brave/brave-core
# Add session expired state to vpn's purchased state

Resolves https://github.com/brave/brave-browser/issues/28811

<img width="376" alt="image" src="https://user-images.githubusercontent.com/6786187/223039332-8dd954b0-5812-467f-8d6c-1124b6c8bbbf.png">

* Toggle button is disabled in session expired state.
* Expired state is cleared after 30days passed.
* Brave Account link in above msg will load https://account.brave.com/account in new tab

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

`BraveVPNServiceTest.LoadPurchasedStateSessionExpiredTest`

1. Enable vpn feature and make purchased state with @brave.com account with promotion
2. Check panel has main view(toggle button)
3. Cancel plan and check `canceled` state from `skus.state` in brave://local-state/
4. Modify os system time more than expired date in `skus.state`
   ex) `skus.state` has `"expires_at\":\"2023-04-09T09:51:57.938456\"` so set system's date to `2023-04-10`
5. Click vpn button and checks above expires view is shown
6. Modify os system's date to `2023-05-11` to invalidate  expired state
7. Click vpn button and check vpn panel shows sell view 

NOTE: When adjust os time, sometimes skus service gives invalid credential summary or 400 error when asking it. Because of this, vpn panel could show `INVALID` state instead of this expired msg. I think this due to manual time manipulation. I assume it'll work well with real world scenario.
